### ISSUE TITLE
[2.0] Error Vbox : has the same UUID

### ISSUE THREAD

**Issue by Raizo62** [GNS3/gns3-gui#1821]
Hi

When i open projects (created by my students), sometimes, i have this error message :

```
VirtualBox has returned an error: VBoxManage: error: Trying to open a VM config '/home/user/projects/TP_SMTP/project-files/virtualbox/b247aee0-fd28-4c20-b0d6-45044a0523ef/Entreprise (Linux)/Entreprise (Linux).vbox' which has the same UUID as an existing virtual machine
VBoxManage: error: Details: code NS_ERROR_FAILURE (0x80004005), component MachineWrap, interface IMachine, callee nsISupports
VBoxManage: error: Context: "OpenMachine(Bstr(a->argv[0]).raw(), machine.asOutParam())" at line 85 of file VBoxManageMisc.cpp
```
[error.log.txt](https://github.com/GNS3/gns3-gui/files/735756/error.log.txt)
[TP_SMTP.gns3.txt](https://github.com/GNS3/gns3-gui/files/735757/TP_SMTP.gns3.txt)
[ListFiles.txt](https://github.com/GNS3/gns3-gui/files/735763/ListFiles.txt)

**Comment by Raizo62** [comment]
an idea this morning : 

(For memory : i use LiveRaizo then, a backup saves the configs of VirtualBox, GNS3 and topology)

Before start GNS3, VirtualBox contains already 2 VBox (of my topology) with the same name :
The VM "CFA (Linux)" with no error
The VM "CFA (Linux)" with the error message of VirtualBox : Inaccessible
and i have the same with 2 other VBox "Entreprise (Linux)"

The bug is perhaps with VMBox named with "(" ?

**Comment by Raizo62** [comment]
Yes is that.

On GNS3, i put a VMBox device with the name "DNS ( Linux )" with space.
GNS3 renames it "DNS(Linux)" on the lab, mais the name in VirtualBox is always "DNS ( Linux )" with space.
When i close GNS3, the VM "DNS ( Linux )" is always in VirtualBox

**Comment by noplay** [comment]
I don't manage to replicate it's unclear for me how you create the issue.  It seem when you use linked clone.

If you start with a blank project what next ? You manually rename one VM ?

**Comment by Raizo62** [comment]
Yes, it is an empty project and i use one linked clone. (I have tested again with GNS3 of yesterday)

**Comment by noplay** [comment]
You just put two linked clone and it doesn't work?

**Comment by Raizo62** [comment]
No. I proof the bug with 1 linked clone.

I have 1 VMBox with the name "DNS ( Linux )" . Be careful to space in the name.
Automatically, GNS3 renames this VM with this new name "DNS ( Linux )". Be careful to no space in the name.
But VirtualBox Interface has always the VM with the old name with space.
When you close GNS3, the VM "DNS ( Linux )" is always in VirtualBox interface.

**Comment by Raizo62** [comment]
Be careful : with this proof, i don't see the errror message "same UUID". But the configuration of VirtualBox/GNS3 is bad. Then the reload of the project does errors "same UUID"

**Comment by noplay** [comment]
When you drag the device you see DNS(Linux) and not DNS(Linux)-1?

**Comment by Raizo62** [comment]
yes : DNS(Linux).
If i use a second VM with the same name 'DNS ( Linux )", GNS3 renames it with "DNS(Linux)1".
But the name in VirtualBox interface is always "DNS ( Linux )"

**Comment by noplay** [comment]
The name _DNS ( Linux )_: with space after the ( is not possible in my 5.1.10 VirtualBox GUI. But possible with 5.1.14. It's work fine on OSX

From Live Raizo website I see you use:
VirtualBox 5.1.4 r110228

For me this version was buggy and could explain a lot of problems, I didn't use Vbox 5.1.X before 5.1.10

**Comment by Raizo62** [comment]
You have an very old release of Live Raizo v8 (beta)
I hope release an stable of v8 this week end, if this bug is corrected (and i don't found another serious error)

In my version, it is VirtualBox 5.1.14 r112924 and the name of VM in virtualBox is with space.

**Comment by noplay** [comment]
And you have issue only when you have " ( "? You can clone this VM in the Vbox GUI without issues?

**Comment by Raizo62** [comment]
here is debug of GNS3
[dns.zip](https://github.com/GNS3/gns3-gui/files/742503/dns.zip)

**Comment by Raizo62** [comment]
I believe that the problem is not clone. I can do TP without problems.
This is as if GNS3 uses original but bad name with space ("DNS ( Linux )") to rename the VM from Debian-1 to "DNS(Linux)"

**Comment by noplay** [comment]
Ok I understand. If you rename a VM to a name that already exists in VirtualBox, Vbox will allow it but after will be lost and will not know VM he need to use.

In GNS3 the rename will half work because the name will not change but under the hood the VM change's name in Virtualbox.

**Comment by Raizo62** [comment]
euh... I am not sure  

>If you rename a VM to a name that already exists in VirtualBox, Vbox will allow it but after will be lost and will not know VM he need to use.

I don't rename a VM to a name that already exists in VirtualBox, because my lab has only VM called "Debian-1". I want rename it to "DNS ( Linux )"

> In GNS3 the rename will half work because the name will not change but under the hood the VM change's name in Virtualbox.

I believe that it the opposite : the name in GNS3 and VirtualBox change. But VirtualBox uses my choosen name "DNS (Linux)" and GNS3 uses the corrected name "DNS(Linux)"

**Comment by Raizo62** [comment]
Perhaps you explain what the code does and then why there are a bug ?

**Comment by noplay** [comment]
Could you test last code? You need to upgrade server and gui.

**Comment by noplay** [comment]
Beware because VirtualBox doesn't protect the usage of two VM with the same name your VirtualBox library could be corrupt. This could be also the problem of: https://github.com/GNS3/gns3-gui/issues/1818

**Comment by Raizo62** [comment]
> Beware because VirtualBox doesn't protect the usage of two VM with the same name your VirtualBox library could be corrupt. This could be also the problem of: #1818

I works always with Live Raizo. Then, normally, my config is always clean after boot.

I compile new version of Live Raizo and i test your change.

**Comment by Raizo62** [comment]
It works. :-D Thanks

**Comment by noplay** [comment]
Thanks for the help does it fix #1818?
